apps/prairielearn/src/pages/instructorAssessmentInstance/instructorAssessmentInstance.js

```diff
@@ -77,14 +77,35 @@ router.get(
         res.locals.assessment_instance.id,
         false,
       );
-
+      const fingerprintNumbers = new Map();
+      let i = 1;
       const stringifier = stringifyStream({
         header: true,
-        columns: ['Time', 'Auth user', 'Event', 'Instructor question', 'Student question', 'Data'],
+        columns: [
+          'Time',
+          'Auth user',
+          'Fingerprint',
+          'IP Address',
+          'Event',
+          'Instructor question',
+          'Student question',
+          'Data',
+        ],
         transform(record) {
+          if (record.client_fingerprint) {
+            if (!fingerprintNumbers.get(record.client_fingerprint.id)) {
+              fingerprintNumbers.set(record.client_fingerprint.id, {
+                client_fingerprint_number: i,
+              });
```

Comment:
Is there any particular reason you're storing an object like `{ client_fingerprint_number: 1 }` instead of just the number? That is, did you consider doing this instead?

```suggestion
              fingerprintNumbers.set(record.client_fingerprint.id, i);
```

Note that you'll need to make other corresponding changing if you want to accept this suggestion.

Reply:
That makes sense. I have updated the map to store just the number instead of an object in commit [777efe4](https://github.com/PrairieLearn/PrairieLearn/pull/9082/commits/777efe4030986da84b7ef0908c85f6ff4de0663a).